jdaviz/core/loaders/importers/spectrum_list/spectrum_list.py

```diff
@@ -34,32 +134,117 @@ def is_valid(self):
         return (isinstance(self.input, (SpectrumList, SpectrumCollection))
                 or (isinstance(self.input, Spectrum) and self.input.flux.ndim == 2))
 
-    @property
-    def output(self):
-        if not self.is_valid:  # pragma: nocover
-            return None
+    def _on_spectra_selected_change(self, change={}):
+        if not len(self.spectra_selected):
+            self.resolver.import_disabled = True
+        else:
+            self.resolver.import_disabled = False
+
+    def _on_format_selected_change(self, change={}):
+        if change['new'] == '1D Spectrum List':
+            self._on_spectra_selected_change()
+        else:
+            self.resolver.import_disabled = False
+
+    @observe('exposures_selected')
+    def _on_exposure_selection_change(self, change={}):
+        """
+        This method is called when the exposure selection changes.
+        It updates the spectra shown to the user based on the selected exposure.
+        """
+        if len(self.exposures_selected) == 0:
+            self.spectra.items = self._spectra_items_helper
+            return
+
+        # Populated with already selected items
+        result = [item for item in self.spectra.items
+                  if item['label'] in self.spectra.selected]
+        # Filter spectra based on the selected exposure
+        for exposure in self.exposures.selected:
+            result.extend(self._exposures_helper.get(exposure, self._spectra_items_helper))
+
+        self.spectra.items = result
```

Comment:
again, if this uses the built-in filter functionality, then we don't need to reproduce that functionality here and `_spectra_items_helper` will probably no longer be needed